iothub/device/src/InternalClient.cs

```diff
@@ -1009,22 +984,8 @@ internal async Task OnMethodCalledAsync(MethodRequestInternal methodRequestInter
             }
             finally
             {
-                try
-                {
-                    methodResponseInternal?.Dispose();
-                    _methodsDictionarySemaphore.Release();
-                }
-                catch (SemaphoreFullException)
-                {
-                    // this semaphore is typically grabbed at the start of the method, but if
-                    // this method is canceled while waiting to grab the semaphore, then this semaphore.release
-                    // will throw this SemaphoreFullException since it was never grabbed in the first place
-                    if (Logging.IsEnabled)
-                    {
-                        Logging.Info(this, "SemaphoreFullException thrown while releasing methods dictionary semaphore," +
-                            " but will be ignored since that means the semaphore is available for other threads to grab again anyways");
-                    }
-                }
+                methodResponseInternal?.Dispose();
+                _methodsDictionarySemaphore.Release();
```

Comment:
Something to consider is that if the call to Dispose throws, then we won't release the semaphore

Reply:
Good point. This will require a small rework to take that into account